terraform/create.go

```diff
@@ -104,6 +125,20 @@ func (t *Terraform) Create() error {
 				return
 			}
 
+			// Upload binary if needed.
+			if uploadBinary {
+				f, err := os.Open(binaryPath)
+				if err != nil {
+					mlog.Error("error opening file", mlog.String("file", binaryPath), mlog.Err(err))
+					return
+				}
+				defer f.Close()
+				if err := sshc.Upload(f, false, "/opt/mattermost/bin/mattermost"); err != nil {
+					mlog.Error("error uploading file", mlog.String("file", binaryPath), mlog.Err(err))
+					return
+				}
```

Comment:
What about making a function out of this? Uploading a file given a path and a `ssh.Client` if we want to keep it here or otherwise even a client method inside `ssh` that only takes the path.

Reply:
Good point. I will do this in https://mattermost.atlassian.net/browse/MM-22977.